### PR TITLE
Node package updates, passing tests

### PR DESCRIPTION
**Pull request by glennreyes**

### CODE REVIEW

**Comment by bvaughn** [comment]
Hey @glennreyes :) Thanks for the dependencies bump!

**Comment by bvaughn** [comment]
Hmm, seeing some regressions with this PR.

Tests are failing for me with:

```
RangeError: Maximum call stack size exceeded
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:44:35)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
...
```

Looks like a Webpack issue reported [here](https://github.com/webpack/webpack/issues/1721) but it has not been resolved unfortunately.

~~Also the local demo app doesn't seem to update the "Output" preview after language or text changes. Only the initial preview.~~ **Edit**: I've found a solution for this.

Any insight?

**Comment by bvaughn** [comment]
I just published v1.0.1 to node after manually testing for regressions. It's unfortunate that the issue between highlight.js and Webpack cause tests to fail but I have too much on my plate at the moment to dig into that further. Would welcome help in that area if you're willing, @glennreyes :)

**Comment by glennreyes** [comment]
Great, thanks @bvaughn! The test fails could have something to do with `"highlight.js": "^9.3.0"` as some class referencing got messed up and `.hljs-variable`disappeared somehow. I would revert the changes for that back to `8.9.1` until we have a clean way to update that 👍

**Comment by bvaughn** [comment]
Unfortunately the issue occurs for 8.9.1 _and_ 9.3.0 so I went ahead and used the latest release. (You can see on isagalaev/highlight.js/issues/895 that the issue was first encountered for version 8.8.0 I think)

**Comment by glennreyes** [comment]
@bvaughn Works everyfine on my machine (with node v6.0.1), hm:

```
07 05 2016 18:12:47.729:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
07 05 2016 18:12:47.735:INFO [launcher]: Starting browser PhantomJS2
07 05 2016 18:12:48.247:INFO [PhantomJS 2.1.1 (Mac OS X 0.0.0)]: Connected on socket /#-_QOJ44wjbDgQoWVAAAA with id 16103533
PhantomJS 2.1.1 (Mac OS X 0.0.0): Executed 1 of 1 SUCCESS (0.006 secs / 0.002 secs)
```

Did some updates on my fork, probably you can check if its good with that version:
https://github.com/glennreyes/react-highlight.js

**Comment by bvaughn** [comment]
Heh, sorry, hadn't pushed my lint-fixed `master` yet :)

Let me try again with a fresh install then and see.

**Comment by glennreyes** [comment]
Let me know if you want another PR ;)

**Comment by bvaughn** [comment]
Not sure what's up. Just did a fresh install (`rm -rf ./node_modules && npm i && npm test`) with `"highlight.js": "^8.9.1"` and I'm still seeing the error:

```
ERROR in ./~/highlight.js/lib/languages/autoit.js
Module parse failed: /Users/brianvaughn/Documents/git/react-highlight.js/node_modules/highlight.js/lib/languages/autoit.js Maximum call stack size exceeded
You may need an appropriate loader to handle this file type.
RangeError: Maximum call stack size exceeded
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:44:35)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
    at Parser.evaluateExpression (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:787:20)
    at Parser.<anonymous> (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/webpack/lib/Parser.js:66:16)
    at Parser.applyPluginsBailResult (/Users/brianvaughn/Documents/git/react-highlight.js/node_modules/tapable/lib/Tapable.js:48:50)
 @ ./~/highlight.js/lib/index.js 13:32-61
Child html-webpack-plugin for "index.html":
```

**Comment by bvaughn** [comment]
For what it's worth, the Webpack issue _does_ seem to hint at the bug being nondeterministic:

> Likewise, we have this problem building vector-im. We have to keep building until we get lucky. It sometimes takes 10 or more runs until it works. Not fun.

So it could just be that you're getting luckier than I am with your test runs.

**Comment by bvaughn** [comment]
I ran it 9 times just now and the 9th worked.

**Comment by glennreyes** [comment]
Hm, interesting. No idea at the moment 😅 Which node version are you running?

**Comment by bvaughn** [comment]
Have tried 5.3.0 and 5.9.1.

**Comment by bvaughn** [comment]
Those are the 2 I've been using with other projects. I'll give 6.1.0 a spin and see if results are any different.

**Comment by bvaughn** [comment]
Cha-ching. 6.1.0 seems to be much less flaky.

**Comment by glennreyes** [comment]
We should consider adding CI like [Travis](travis-ci.org) into this module the keep every desired node versions on track 😎

**Comment by bvaughn** [comment]
Yeah, that would be worth doing. Until now I didn't realize anyone but me was using this particular module so I haven't given it much love. ;)

**Comment by glennreyes** [comment]
I've been looking for some cool highlight syntax module for react and that one was the most funny one to hack 😎

**Comment by bvaughn** [comment]
Interesting to note that tests failed for Node 5.3.0 and 5.9.1 with highlight.js 8.9.1 and 9.3.0. Tests pass for Node 6.1.0 with both versions.

**Comment by glennreyes** [comment]
You sure? It's failing at the moment:

```
PhantomJS 2.1.1 (Mac OS X 0.0.0) Highlight should render content with basic markup FAILED
    expected '' to contain 'hljs'
    assert
    contain
```

That looks now wrong to me imo:

``` js
expect(highlightDOMNode.className).to.contain('hljs')
```

**Comment by bvaughn** [comment]
Sorry for the ambiguity. I'm talking about my local git repo. I haven't pushed everything.

**Comment by bvaughn** [comment]
(I'm in a Starbucks this morning with a slow wifi.) Just pushed.

**Comment by glennreyes** [comment]
Nice, thanks! 😎 Hope the coffee was great

**Comment by glennreyes** [comment]
Having issues now using this module:
image

Try to fix the react refs

**Comment by bvaughn** [comment]
Sorry but I'm not seeing this error locally or on the demo site: http://bvaughn.github.io/react-highlight.js/.

If you can reproduce it in something that you can share with me (show me) then file a bug and I can take a look at it.

**Comment by bvaughn** [comment]
PS. Not to point out the obvious, but do you have multiple copies of React loaded?

**Comment by glennreyes** [comment]
@bvaughn It occurs only when you import this module from another project. We should point the "main" property in `package.json` directly to the React component instead to `dist/main.js`

**Comment by bvaughn** [comment]
Ah, the problem is that package is pointing at a UMD build huh. That's not new with this release.

**Comment by bvaughn** [comment]
Filed issue #3 for this. Will try to get to it at some point. Busy this afternoon working on an upcoming react-virtualized release though.

**Comment by glennreyes** [comment]
Ah cool, thanks! But we've set `libraryTarget: 'commonjs2'` as output in webpack config, right?

**Comment by bvaughn** [comment]
Sorry, I misspoke. I was doing too many things at once over the weekend trying to finish up react-virtualized 7 and doing a js-search release also. I _believe_ the problem is that React isn't configured as an external and so Webpack is dumping it into the dist as well.

**Comment by bvaughn** [comment]
Just published 1.0.2 with React and ReactDOM added to Webpack externals so they didn't get inlined. Give it another spin.

**Comment by glennreyes** [comment]
Wow that sounds awesome! I'll give it a spin this evening.
